tests/test_git.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Pranjal Aswani <aswani.pranjal@gmail.com>
+
+
+import sys
+import unittest
+from datetime import datetime
+
+sys.path.insert(0, '..')
+
+import pandas as pd
+from numpy.testing import assert_array_equal
+from elasticsearch import Elasticsearch
+
+from manuscripts2.metrics import git
+from manuscripts2.elasticsearch import Query, Index, get_trend
+from utils import setup_data_source, teardown_data_source, DATA_SOURCES
+
+
+ES_URL = "http://localhost:9200"
+
+# All values as seen on 2018-07-10
+TREND_LAST = 12
+TREND_PRECENTAGE = -41
+
+# files
+AUTHORS_BY_PERIOD = "data/git_authors_by_months.csv"
+
+
+class TestGit(unittest.TestCase):
+    """
+    Test the git data source
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup necessary infrastructure to test the functions
+        """
+
+        cls.git = DATA_SOURCES['git']
+        setup_data_source('git', cls.git[0], cls.git[1])
+
+    def setUp(self):
+        """
+        Set up the necessary functions to run unittests
+        """
+
+        self.es = Elasticsearch(ES_URL)
+        self.git = DATA_SOURCES['git']
+        self.git_index = Index(self.git[1])
+
+        Query.interval_ = "month"
```

Comment:
This is the kind of behavior I was commenting with respect to initializing properties in a class. Since you're not calling report here, you need to do this initialization. But a casual user, very unlikely will be aware of this *large* effect of calling report...

Reply:
I set the interval once again here, sorry.  I am trying to set it only once in the report.py file as you pointed out above.